libraries/primitive_consul_lock.rb

```diff
@@ -74,7 +75,9 @@ def wait_until(action, opts = {})
       dc = "(in #{@options[:datacenter]})" if @options[:datacenter]
       Chef::Log.info "Will #{action} the lock #{path} #{dc}"
       start = Time.now
-      success = 0.upto(opts[:max_failures] || Float::INFINITY).any? do |tries|
+      max_failures = opts[:max_failures] || @options[:max_failures]
+      raise_on_failures = opts[:raise_on_failures].nil? ? @options[:raise_on_failures] : opts[:raise_on_failures]
```

Comment:
NIT: the `.fetch` method is great for that e.g. `opts.fetch(:raise_on_failures, @options[:raise_on_failures])`